source/joker_effects.c

```diff
@@ -609,13 +761,87 @@ static JokerEffect brainstorm_joker_effect(Joker *joker, Card *scored_card) {
     if (first_joker != NULL && first_joker->joker->id != JOKER_BRAINSTORM_ID) {
         // Static var to avoid infinite blueprint + brainstorm loops
         in_brainstorm = true;
-        effect = joker_get_score_effect(first_joker->joker, scored_card);
+        effect = joker_get_score_effect(first_joker->joker, scored_card, scored_when);
         in_brainstorm = false;
     }
 
     return effect;
 }
 
+
+__attribute__((unused))
+static JokerEffect hack_joker_effect(Joker *joker, Card *scored_card, int scored_when) {
+    JokerEffect effect = {0};
+
+    switch (scored_when) {
+        case JOKER_CALLBACK_ON_HAND_SCORED_END:
+            joker->data = -1;
+            break;
+        
+        case JOKER_CALLBACK_ON_CARD_SCORED:
+            // Works the same way as Mime, but for played cards and check what rank the card is
+            switch (scored_card->rank) {
+                case TWO:
+                case THREE:
+                case FOUR:
+                case FIVE:
+                    effect.retrigger = (joker->data < get_scored_card_index());
+                    joker->data = get_scored_card_index();
+            }
+            
+            break;
+    }
+
+    return effect;
+}
+
+__attribute__((unused))
+static JokerEffect seltzer_joker_effect(Joker *joker, Card *scored_card, int scored_when) {
+    JokerEffect effect = {0};
+
+    switch (scored_when) {
+        case JOKER_CALLBACK_ON_HAND_SCORED_END:
+            joker->halves.data0  = -1;
+            joker->halves.data1 -=  1;
+            if (joker->halves.data1 <= 0) {
+                effect.expire = true;
+                snprintf(effect.message, 8, "Expire!");
```

Comment:
```suggestion
                snprintf(effect.message, JOKER_MSG_BUF_LEN, "Expire!");
```

Batch - Add JOKER_MSG_BUF_LEN